template/razor_test.go

```diff
@@ -504,3 +504,59 @@ func TestMultilineStringProtect(t *testing.T) {
 		})
 	}
 }
+
+func TestData(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name string
+		code string
+		want string
+		err  error
+	}{
+		{"Empty", `@data("")`, "", nil},
+		{"Integer", `@data("1")`, "1", nil},
+		{"Hcl", `@data("a = 1 b = 2")`, "a=1 b=2", nil},
+		{"Hcl type", `@typeOf(data("a = 1 b = 2"))`, "hcl.hclDict", nil},
+		{"Hcl kind", `@kindOf(data("a = 1 b = 2"))`, "map", nil},
+		{"Invalid", "@typeOf(data(`\"a\": 1, \"b\": 2`))", `"<RUN_ERROR>"`, fmt.Errorf("")},
+		{"Json", "@typeOf(data(`{\"a\": 1, \"b\": 2}`))", "json.jsonDict", nil},
+		{"Yaml", "@typeOf(data(`a: 1\nb: 2`))", "yaml.yamlDict", nil},
+		{"Flexible Hcl", "@typeOf(data(`a = 1 b = hello`))", "yaml.yamlDict", nil}, // TODO: Change that to hcl
```

Comment:
TODO for now or later?

Reply:
Shit, I really don't remember. I merged an old branch that has never been submitted as a PR.
I would say later. I think that I wrote that test when I started to look at HCL2 when it was still in Beta IIRC.